packages/marshal/NEWS.md

```diff
@@ -1,5 +1,13 @@
 User-visible changes in `@endo/marshal`:
 
+# next release
+
+- JavaScript's relational comparison operators like `<` compare strings by lexicographic UTF16 code unit order, which is exposes an internal representational detail not relevant to the string's meaning as a Unicode string. Previously, `compareRank` and associated functions compared strings using this JavaScript-native comparison. Now `compareRank` and associated functions compare strings by lexicographic Unicode Code Point order. ***This change only affects strings containing so-called supplementary characters, i.e., those whose Unicode character code does not fit in 16 bits***.
+  - This release does not change the `encodePassable` encoding. But now, when we say it is order preserving, we need to be careful about which order we mean. `encodePassable` is rank-order preserving when the encoded strings are compared using `compareRank`.
```

Comment:
@gibson042 is this true? It was true for my small test case, which proves very little. Will the same property also be true for `compactOrdered`? For either, does restricting these strings to well-ordered have any effect on whether their encoding is order preserving?

Reply:
Good point, and bad news!

Grepping for `.sort()` specifically with nothing between the parens, I see 96 occurrences in agoric-sdk and 26 in endo. Some may not be or contain strings. But still, fixing all that do will be disruptive. And the longer we wait, the more disruptive it'll be.

I'm putting this back into Draft until we decide what our plan is. Attn @ivanlei

Reply:
Is there any practical way to scan a recent snapshot of our chain and ***somehow*** see how many persistent strings are
- non-ascii,
- non-well-formed, or
- have supplementary characters (those whose code is > 16 bits)

? 
How hard would it be?
Attn @mhofman 

NOT URGENT.